invoices/invoiceregistry.go

```diff
@@ -640,7 +640,10 @@ func (i *InvoiceRegistry) CancelInvoice(payHash lntypes.Hash) error {
 		canceledHtlcs := make(
 			map[channeldb.CircuitKey]*channeldb.HtlcAcceptDesc,
 		)
-		for key := range invoice.Htlcs {
+		for key, htlc := range invoice.Htlcs {
+			if htlc.State == channeldb.HtlcStateCancelled {
```

Comment:
Does this mean that we can cancel a settled one?

Reply:
Discussed offline. No htlc should be in the `settled` state here, because then the invoice state would have been `settled`. Added extra check to return an error if this happens.